src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -181,22 +194,29 @@ private void loadRef(Map<String, String> headers, String endPoint, Handler<Async
       Future f = Future.future();
       futures.add(f);
 
-      HttpClientRequest req = httpClient.postAbs(endPointUrl, x -> {
-        if (x.statusCode() >= 200 && x.statusCode() <= 299) {
+      JsonObject jsonObject = new JsonObject(json);
+      String id = jsonObject.getString("id");
+      if (id == null) {
+        f.handle(Future.failedFuture("Missing id for " + json));
+        return;
+      }
+      HttpClientRequest reqPut = httpClient.putAbs(endPointUrl + "/" + id, resPut -> {
+        if (resPut.statusCode() == 404) {
+          HttpClientRequest reqPost = httpClient.postAbs(endPointUrl, resPost -> {
+            if (resPost.statusCode() == 201) {
+              f.handle(Future.succeededFuture());
+            } else {
+              f.handle(Future.failedFuture("POST " + endPointUrl + " returned status " + resPost.statusCode()));
+            }
+          });
+          applyHeaders(reqPost, headers, json);
+        } else if (resPut.statusCode() == 200) {
           f.handle(Future.succeededFuture());
         } else {
-          f.handle(Future.failedFuture("POST " + endPointUrl + " returned status " + x.statusCode()));
+          f.handle(Future.failedFuture("PUT " + endPointUrl + "/" + id + " returned status " + resPut.statusCode()));
```

Comment:
Using guard clauses can reduce the nesting making it more easy to read and understand:

        if (resPut.statusCode() == 200) {
          f.handle(Future.succeededFuture());
          return;
        }
        if (resPut.statusCode() != 404) {
          f.handle(Future.failedFuture("PUT " + endPointUrl + "/" + id + " returned status " + resPut.statusCode()));
          return;
        }
        // resPut.statusCode() == 404
        HttpClientRequest reqPost = httpClient.postAbs(endPointUrl, resPost -> {
          if (resPost.statusCode() == 201) {
            f.handle(Future.succeededFuture());
          } else {
            f.handle(Future.failedFuture("POST " + endPointUrl + " returned status " + resPost.statusCode()));
          }
        });
        applyHeaders(reqPost, headers, json);

Reply:
I see, but it would break.. The f.handle stuff is collected and a return would therefore not handle "res" as it should. Your comment revealed a mistake that I did however :-)